dataset/src/main/scala/frameless/functions/Lit.scala

```diff
@@ -8,52 +8,18 @@ import org.apache.spark.sql.types.DataType
 private[frameless] case class Lit[T <: AnyVal](
     dataType: DataType,
     nullable: Boolean,
-    toCatalyst: CodegenContext => ExprCode,
-    show: () => String
+    show: () => String,
+    catalystExpr: Expression // must be a generated Expression from a literal TypedEncoder's toCatalyst function
 ) extends Expression with NonSQLExpression {
   override def toString: String = s"FramelessLit(${show()})"
 
-  def eval(input: InternalRow): Any = {
-    val ctx = new CodegenContext()
-    val eval = genCode(ctx)
-
-    val codeBody = s"""
-      public scala.Function1<InternalRow, Object> generate(Object[] references) {
-        return new LiteralEvalImpl(references);
-      }
-
-      class LiteralEvalImpl extends scala.runtime.AbstractFunction1<InternalRow, Object> {
-        private final Object[] references;
-        ${ctx.declareMutableStates()}
-        ${ctx.declareAddedFunctions()}
-
-        public LiteralEvalImpl(Object[] references) {
-          this.references = references;
-          ${ctx.initMutableStates()}
-        }
-
-        public java.lang.Object apply(java.lang.Object z) {
-          InternalRow ${ctx.INPUT_ROW} = (InternalRow) z;
-          ${eval.code}
-          return ${eval.isNull} ? ((Object)null) : ((Object)${eval.value});
-        }
-      }
-    """
-
-    val code = CodeFormatter.stripOverlappingComments(
-      new CodeAndComment(codeBody, ctx.getPlaceHolderToComments())
-    )
-
-    val (clazz, _) = CodeGenerator.compile(code)
-    val codegen =
-      clazz.generate(ctx.references.toArray).asInstanceOf[InternalRow => AnyRef]
-
-    codegen(input)
-  }
```

Comment:
I think this code was left to deal with some serialization issues long time ago. 

I was able to find the old commit that introduced it: https://github.com/typelevel/frameless/blame/59caada7955c90fce266577109122265b7840418/dataset/src/main/scala/frameless/functions/Lit.scala 

Let's not remove this code, for now?

Reply:
Serialization test failures could be tricky to find, i'd vote for a separate PR with investigations ~ mb indeed we don't need this portion of code anymore.